staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -177,45 +212,73 @@ func newTestableController(config TestableConfig) *configController {
+				cfgCtlr.finishHandlingNotification(cfgCtlr.configQueue, newObj)
 			}
 		},
 		DeleteFunc: func(obj interface{}) {
 			name, _ := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
 			klog.V(7).Infof("Triggered API priority and fairness config reloading due to deletion of PLC %s", name)
-			cfgCtlr.configQueue.Add(0)
+			cfgCtlr.finishHandlingNotification(cfgCtlr.configQueue, obj)
 
 		}})
 	fsi.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			fs := obj.(*flowcontrol.FlowSchema)
 			klog.V(7).Infof("Triggered API priority and fairness config reloading due to creation of FS %s", fs.Name)
-			cfgCtlr.configQueue.Add(0)
+			cfgCtlr.finishHandlingNotification(cfgCtlr.configQueue, obj)
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
 			newFS := newObj.(*flowcontrol.FlowSchema)
 			oldFS := oldObj.(*flowcontrol.FlowSchema)
-			if !apiequality.Semantic.DeepEqual(oldFS.Spec, newFS.Spec) {
-				klog.V(7).Infof("Triggered API priority and fairness config reloading due to spec update of FS %s", newFS.Name)
-				cfgCtlr.configQueue.Add(0)
+			// Changes to either Spec or Status are relevant.  The
+			// concern is that we might, in some future release, want
+			// different behavior than is implemented now. One of the
+			// hardest questions is how does an operator roll out the
+			// new release in a cluster with multiple kube-apiservers
+			// --- in a way that works no matter what servers crash
+			// and restart when. If this handler reacts only to
+			// changes in Spec then we have a scenario in which the
+			// rollout leaves the old Status in place. The scenario
+			// ends with this subsequence: deploy the last new server
+			// before deleting the last old server, and in between
+			// those two operations the last old server crashes and
+			// recovers. The chosen solution is making this controller
+			// insist on maintaining the particular state that it
+			// establishes.
+			if !(apiequality.Semantic.DeepEqual(oldFS.Spec, newFS.Spec) &&
+				apiequality.Semantic.DeepEqual(oldFS.Status, newFS.Status)) {
+				klog.V(7).Infof("Triggered API priority and fairness config reloading due to spec and/or status update of FS %s", newFS.Name)
+				cfgCtlr.finishHandlingNotification(cfgCtlr.configQueue, newObj)
 			}
 		},
 		DeleteFunc: func(obj interface{}) {
 			name, _ := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
 			klog.V(7).Infof("Triggered API priority and fairness config reloading due to deletion of FS %s", name)
-			cfgCtlr.configQueue.Add(0)
+			cfgCtlr.finishHandlingNotification(cfgCtlr.configQueue, obj)
 
 		}})
 	return cfgCtlr
 }
 
+// enqueueEverything enqueues an item that causes processing of all
+// the config objects and is the normal way to respond to a
+// notification about a config object.  Put this in
+// TestableConfig.FinishHandlingNotification to get normal behavior.
+func enqueueEverything(wq workqueue.RateLimitingInterface, obj interface{}) {
```

Comment:
nit: change `obj` to `_` to make it explicit that the parameter is not used

Reply:
done